src/runtime/TemporalObject.cpp

```diff
@@ -68,228 +68,268 @@ Value TemporalObject::toISOTime(ExecutionState& state, DateObject& d)
+
+            if (isoString[index] == '.' || isoString[index] == ',') {
+                std::map<std::string, int> seconds;
+                seconds = TemporalObject::getSeconds(isoString, index);
+                dateTime.millisecond = seconds["milliSecond"];
+                dateTime.microsecond = seconds["microSecond"];
+                dateTime.nanosecond = seconds["NanoSecond"];
+            }
+        }
+
+        // Offset
+        if (isoString[index] == 'z' || isoString[index] == 'Z') {
+            ++index;
+        } else if (isoString[index] == '+' || isoString[index] == '-' || isoString.rfind("−", index) == 0) {
+            TemporalObject::offset(isoString, index);
+        }
+
+        // TimeZone
+        if (isoString[index] != 'E') {
+            while (TemporalObject::tzComponent(isoString, index).length() != 0)
+                ;
+        } else if (isoString.rfind("Etc/GMT", index) == 0) {
+            isoString = isoString.substr(index + 7);
+            if (isoString[index] == '+' || isoString[index] == '-') {
+                if (!std::isdigit(isoString[index + 1])) {
+                    throw std::exception();
+                }
+
+                if (is_number(isoString.substr(index + 1, 2))) {
+                    index += 3;
+                }
+                index += 2;
+            }
+        } else if (isoString[index] == '+' || isoString[index] == '-' || isoString.rfind("−", index) == 0) {
+            TemporalObject::offset(isoString, index);
+        }
+
+        // Calendar
+        if (isoString.rfind("[u-ca=") == 0) {
+            index += 6;
+            for (int i = 0; isoString[index] != ']'; ++i) {
+                // Check if current char is a number or
+                if (std::isalpha(isoString[index]) || std::isdigit(isoString[index])) {
+                    dateTime.calendar += isoString[index];
+                    ++index;
+                }
+
+                // A calendar can't be longer than 8 character
+                if (i == 8) {
+                    if (isoString[index] != '-' && isoString[index] != ']') {
+                        throw std::exception();
+                    }
+                    i = 0;
+                }
+
+                // A calendar must be at least 3 character
+                if (i < 3 && (isoString[index] == '-' || isoString[index] == ']')) {
+                    throw std::exception();
+                }
+            }
+            if (isoString[index] == ']') {
+                index++;
             }
-        } else if (date.length() > 7) {
-            goto good;
         }
-    }
 
-    if (!extra.empty()) {
-        size_t n = std::count(extra.begin(), extra.end(), '[');
-        if (n == 1 || n == 2) {
-            goto good;
+        if (isoString.length() != index) {
+            throw std::exception();
         }
+    } catch (std::exception& e) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid ISO string");
```

Comment:
It seems that the exception value (e) is not mandatory and nowhere used, so please replace this try-catch statement with simple if-else statement